x-pack/plugins/lens/server/embeddable/make_lens_embeddable_factory.ts

```diff
@@ -155,6 +156,17 @@ export const makeLensEmbeddableFactory =
                   attributes: migratedLensState,
                 } as unknown as SerializableRecord;
               },
+              '8.6.0': (state) => {
+                const lensState = state as unknown as {
```

Comment:
nit: it would be slightly more correct to use `state as unknown as SavedObject<LensDocShape850<VisState850>>`

Otherwise, judging by the types alone, it would be correct to just return `{attributes: migratedLensState}` because the types ignore that a saved object has an `id` and all other root fields.

This is outside of the scope of this PR, but I think actually it would be better if the embeddable migration types allowed you to use `migrateIndexPatternDatasource` because apart from the type incompatibility the embeddable and saved object migrations are identical.